### ISSUE TITLE
json_tags `camel` should be `camelGo`

### ISSUE THREAD

**Issue by gnuletik** [sqlc-dev/sqlc#1585]
### Version

Other

### What happened?

Following [this PR](https://github.com/kyleconroy/sqlc/pull/905), it is now possible to configure json_tags case style with the following config:

```
  "emit_json_tags": true,
  "json_tags_case_style": "camel"
```

However, the `camel` option will generate JSON field `itemID` for an SQL column `item_id`.
To match camel case, it should be `itemId`.
I think that we should keep the current behavior under another name, like `camelGo` or `camelInitialism`.

However, this may introduce a breaking change for existing projects, so it may be wiser to add a new value for `json_tags_case_style` like `camelJson`.

The issue seems to be here :

https://github.com/kyleconroy/sqlc/blob/aade08708ea0c4aa88fe7b5dbe929e325c323ccd/internal/codegen/golang/field.go#L57-L79

### Playground URL

https://play.sqlc.dev/p/e8395fd011f50b964b6356488bea183153ceb76ab74c140841dcd3563a517cc2

**Comment by kyleconroy** [comment]
Yep, I think this is totally fair and was probably a mistake on our part.

**Comment by kyleconroy** [comment]
Fixed as of 1.21 

```go
package db

import (
	"database/sql"
)

type Author struct {
	ID       int64          `json:"id"`
	LinkedID sql.NullInt64  `json:"linkedId"`
	Name     string         `json:"name"`
	Bio      sql.NullString `json:"bio"`
}
```

https://play.sqlc.dev/p/431b402ab453f262fe914db9c617426f400a18fd02a0d840a77d1d84c8270d24

**Comment by gnuletik** [comment]
Thanks for the fix @kyleconroy !

Would it be possible to keep the previous behavior under the `"json_tags_case_style": "camelGo"` configuration?

As we are using multiple sqlc generated types in API responses, the 1.21 upgrade would introduce a breaking change in the API.
